### PR TITLE
[18.0-fr1] This patch allows Octavia to set its route annotations.

### PR DESCRIPTION
**Pull request by fernandoroyosanchez**
Depends-On:
https://github.com/openstack-k8s-operators/octavia-operator/pull/438 
Related-Issue: [OSPRH-10959](https://issues.redhat.com//browse/OSPRH-10959)
(cherry picked from commit f91c5f8)

### CODE REVIEW

**Comment by fernandoroyosanchez** [comment]
kuttl failing looks unrelated...
/retest

**Review by fao89 — APPROVED** [review]

File: apis/go.mod [hunk 1]
```diff
@@ -22,7 +22,7 @@ require (
 	github.com/openstack-k8s-operators/mariadb-operator/api v0.5.0
 	github.com/openstack-k8s-operators/neutron-operator/api v0.5.1-0.20241112143215-294abc29ae85
 	github.com/openstack-k8s-operators/nova-operator/api v0.5.0
-	github.com/openstack-k8s-operators/octavia-operator/api v0.5.0
+	github.com/openstack-k8s-operators/octavia-operator/api v0.5.1-0.20241119144914-248c7edb4442
```
Comment (gthiemonge) [inline apis/go.mod:25]: shouldn't it be 0.5.0-something? how do we check that this version is the right one?

Comment (fernandoroyosanchez) [reply apis/go.mod:25]: I get from: 

```
froyo@fedora ~ $ go list -m -json github.com/openstack-k8s-operators/octavia-operator/api@18.0-fr1 | jq '.|"\(.Path) \(.Version)"'
"github.com/openstack-k8s-operators/octavia-operator/api v0.5.1-0.20241119144914-248c7edb4442"
```

**Pull request merged**